File: ProblemSetsEasy/solve/javascript/ProblemSetsEasySolver.js
#! /usr/bin/env node

var fs = require("fs");
var path = require("path");
var tc = null;

function init() {
    var gettcHome = process.env.GETTC_HOME || path.join(process.env.HOME, ".gettc");
    var includeDir = path.join(gettcHome, "include", "javascript");
    tc = require(path.join(includeDir, "topcoder"));
}

function main() { try {
    var input = fs.readFileSync(process.argv[2], { encoding: "ascii" });
    var reader = new tc.Reader(input);
    var E = reader.next("int"); reader.next();
    var EM = reader.next("int"); reader.next();
    var M = reader.next("int"); reader.next();
    var MH = reader.next("int"); reader.next();
    var H = reader.next("int");

    var ProblemSetsEasy = require("./ProblemSetsEasy");
    var result = ProblemSetsEasy.maxSets(E, EM, M, MH, H);
    
    var writer = new tc.Writer();
    writer.next(result, "int");
    fs.writeFileSync(process.argv[3], writer.toString(), { encoding: "ascii" });
} catch (err) {
    console.log(err.toString());
    console.trace();
}}

init();
main();

File: ProblemSetsEasy/solve/java/ProblemSetsEasySolver.java
import java.io.FileReader;
import java.io.FileWriter;

import java.util.List;
import java.util.ArrayList;

import org.topcoder.TopcoderReader;
import org.topcoder.TopcoderWriter;
import org.topcoder.TypeRef;

public class ProblemSetsEasySolver {
    public static void main(String[] args) {
    try {
        TopcoderReader reader = new TopcoderReader(new FileReader(args[0]));
        int E = (Integer) reader.next(Integer.class);
        reader.next();
        
        int EM = (Integer) reader.next(Integer.class);
        reader.next();
        
        int M = (Integer) reader.next(Integer.class);
        reader.next();
        
        int MH = (Integer) reader.next(Integer.class);
        reader.next();
        
        int H = (Integer) reader.next(Integer.class);
        reader.close();

        ProblemSetsEasy solver = new ProblemSetsEasy();
        TopcoderWriter writer = new TopcoderWriter(new FileWriter(args[1]));
        writer.write(solver.maxSets(E, EM, M, MH, H));
        writer.close();
    } catch (Exception err) {
        err.printStackTrace(System.err);
    }
    }    
}
